docs/codeql/codeql-cli/creating-codeql-databases.rst

```diff
@@ -228,6 +228,113 @@ commands that you can specify for compiled languages.
   This command runs a custom script that contains all of the commands required
   to build the project.
 
+Using indirect build tracing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the CodeQL CLI autobuilders for compiled languages do not work with your CI workflow and you cannot specify
+build commands, you can use indirect build tracing to create a CodeQL database. To use indirect build tracing, your CI system must be able to set custom environment variables for each build action.
```

Comment:
I would swap _"cannot specify build commands"_ with _"cannot wrap invocations of build command with `codeql database trace-command`"_. The build commands will be specified somewhere, it's just a question of how practical it would be for the user to put `codeql database trace-command` in front of those build commands.